nidirect_common/inc/landing_page.redirects.inc

```diff
@@ -0,0 +1,129 @@
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\Language;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\node\Entity\Node;
+use Drupal\redirect\Entity\Redirect;
+
+/**
+ * Validation handler for landing page node forms.
+ *
+ * Checks whether there is already a landing page associated with the theme selected.
+ *
+ * @param array $form
+ *   The form array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state collection.
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function nidirect_common_landing_page_form_validate(array &$form, FormStateInterface $form_state) {
+  $theme_selected = $form_state->getValue('field_subtheme')[0]['target_id'];
+
+  $query = \Drupal::entityTypeManager()->getStorage('node')->getQuery();
+  $result = $query->condition('type', 'landing_page', '=')
+    ->condition('field_subtheme', $theme_selected, '=')
+    ->execute();
+
+  if (!empty($result)) {
+    $term = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->load($theme_selected);
+    $node = \Drupal::entityTypeManager()->getStorage('node')->load(array_pop($result));
+    $link = Link::createFromRoute($node->label(), 'entity.node.edit_form', ['node' => $node->id()])->toRenderable();
+
+    $message = new TranslatableMarkup("There is already a landing page node (@landing_page) associated with '@term_name'", [
+      '@landing_page' => \Drupal::service('renderer')->render($link),
+      '@term_name' => $term->label()
+    ]);
+
+    $form_state->setErrorByName('field_subtheme', $message);
+  }
+}
+
+/**
+ * Function to re-use checks for landing page node with valid theme field selection.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   The entity to evaluate.
+ * @return bool
+ *   TRUE if it is a landing page with theme field value, otherwise FALSE.
+ */
+function nidirect_common_is_landing_page_with_theme_value(EntityInterface &$entity) {
+  if ($entity instanceof Node == FALSE) {
+    return FALSE;
+  }
+
+  $has_subtheme_value = $entity->hasField('field_subtheme') && !empty($entity->get('field_subtheme')->getString());
+
+  // Chain logic conditions: if one condition fails any remaining checks are skipped and the expression returns FALSE.
+  return ($entity->bundle() == 'landing_page' && $has_subtheme_value);
+}
+
+/**
+ * Detect whether a redirect already exists, return a URL of the
+ * destination if it does.
+ *
+ * @param string $source_path
+ *   The source path.
+ * @return string
+ *   Redirect URL if found, empty string if not.
+ */
+function nidirect_common_get_redirect_url(string $source_path) {
+  $redirect = \Drupal::service('redirect.repository')->findBySourcePath($source_path);
+
+  if (!empty(($redirect))) {
+    $redirect = array_pop($redirect);
+    $redirect_url = $redirect->getRedirectUrl()->toString();
+  }
+  else {
+    $redirect_url = '';
```

Comment:
I'd assign this var at the top and then you can do away with the else statement.

Reply:
Yep, agreed. Like this